source/sync/data-access-patterns/flexible-sync-permissions-guide.txt

```diff
@@ -0,0 +1,607 @@
+  select the template called "Users can only read and write their own data".
+
+  This populates the rule expression box with something like the following:
+
+  .. code-block:: json
+     :emphasize-lines: 6-7
+  
+     {
+       "defaultRoles": [
+         {
+           "name": "owner-read-write", 
+           "applyWhen": {},
+           "read": {"owner_id": "%%user.id"},
+           "write": {"owner_id": "%%user.id"}
+         }
+       ]  
+     }
+
+  Note that the "read" and "write" expressions use the ``owner_id`` field we
+  marked as "queryable" above. It also uses the ``%%user`` "expansion" to read
+  the requesting user's id. Realm replaces (or "expands") the expansion at
+  evaluation time with the corresponding value -- in this case, the user object.
+
+  When a client accesses the data through Sync, Realm evaluates these
+  expressions to determine permissions. This expression says the requesting user
+  may read or write if the document's ``owner_id`` field is equal to the user's
+  ID.
+
+- Skip :guilabel:`Advanced Configuration`, and then click :guilabel:`Save Changes` 
+  to enable Sync with these permissions.
+
+
+.. _write-own-data-read-all-data:
+
+Write Own Data, Read All Data
+-----------------------------
+
+In this case, users can read all data, but write only their own data. Consider a 
+recipe app where users can read all recipes and add new recipes. The recipes 
+they add can be viewed by everyone using the app. 
+
+To set this up, follow these steps:
+
+.. include:: /includes/use-flex-sync-templates
+
+- :guilabel:`Define Permissions`: use the template dropdown to
+  select the template called "Users can read all data but only write their own data".
+
+  This populates the rule expression box with the following:
+
+  .. code-block:: json
+     :emphasize-lines: 6-7
+  
+     {
+        "defaultRoles": [
+          {
+            "name": "owner-write", 
+            "applyWhen": {},
+            "read": true,
+            "write": {"owner_id": "%%user.id"}
+          }    
+        ]  
+      }
+
+  Note that the "read" expression is set to ``true``, indicating that no matter
+  which user is authenticated, they can access all of the data. The "write"
+  expression uses the ``owner_id`` field we marked as "queryable" above and uses
+  the ``%%user`` "expansion" to match against the requesting user's id. Realm
+  replaces (or "expands") the expansion at evaluation time with the
+  corresponding value -- in this case, the user object.
+
+  When a user accesses the data through Sync, Realm evaluates these
+  expressions to determine permissions. This expression says the requesting user
+  may read all data, but write only if the document's ``owner_id`` field is equal 
+  to the user's ID.
+
+- Skip :guilabel:`Advanced Configuration`, and then click :guilabel:`Save Changes` 
+  to enable Sync with these permissions.
+
+.. _admin-privileges:
+
+Administrator Privileges
+------------------------
```

Comment:
Suggestion/thought: we might want to point out in this section that users should NOT have write access to their own Custom User Data objects... because if so, a malicious user could give themselves admin permissions and access all data in the app very easily.